src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Form/Entity/SamlEntityType.php

```diff
@@ -323,6 +325,18 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
 
             ->add('publishButton', SubmitType::class, ['label'=> $options['publish_button_label'], 'attr' => ['class' => 'button']])
             ->add('cancel', SubmitType::class, ['attr' => ['class' => 'button']]);
+
+        // When the SAML2.0 entity is set to have an UNSPECIFIED name id format (in manage) do not show the field on the form
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event): void {
+            /** @var SaveSamlEntityCommand $data */
+            $data = $event->getData();
+            if ($data->getNameIdFormat() === Constants::NAME_ID_FORMAT_UNSPECIFIED) {
+                $form = $event->getForm();
+                if ($form->has('metadata') && $form->get('metadata')->has('nameIdFormat')) {
+                    $form->get('metadata')->remove('nameIdFormat');
+                }
+            }
+        });
     }
```

Comment:
I see some repetition

Reply:
It's not realy a dry thing in my opinion; the one block deals with the nameIdFormat and the other with the clientId. They kind-of are the same thing. Leaving it as is for now